autoload/go/template.vim

```diff
@@ -10,12 +11,19 @@ function! go#template#create() abort
   let l:package_name = go#tool#PackageName()
 
   " if we can't figure out any package name(no Go files or non Go package
-  " files) from the directory create the template
-  if l:package_name == -1
+  " files) from the directory create the template or use the cwd
+  " as the name
+  if l:package_name == -1 && l:go_template_no_file != 1
     let l:template_file = get(g:, 'go_template_file', "hello_world.go")
     let l:template_path = go#util#Join(l:root_dir, "templates", l:template_file)
     exe '0r ' . fnameescape(l:template_path)
     $delete _
+  elseif l:package_name == -1 && l:go_template_no_file == 1
+    " cwd is now the dir of the package
+    let l:path = fnamemodify(getcwd(), ':t')
+    let l:content = printf("package %s", l:path)
+    call append(0, l:content)
+    $delete _
```

Comment:
Per my comment above, I think you don't need three separate clauses. The following is sufficient:

```viml
  if l:go_template_no_file != 1 || l:package_name == -1
    let l:template_file = get(g:, 'go_template_file', "hello_world.go")
    let l:template_path = go#util#Join(l:root_dir, "templates", l:template_file)
    exe '0r ' . fnameescape(l:template_path)
    $delete _
  else
    let l:content = printf("package %s", l:package_name)
    call append(0, l:content)
    $delete _
  endif
```